CMakeLists.txt

```diff
@@ -96,7 +63,10 @@ endif()
 option(BUILD_EXAMPLES "Build examples" ON)
 if (BUILD_EXAMPLES)
   if (UNIX)
-      add_subdirectory(examples)
+      add_subdirectory(examples/master)
+  endif()
+  if (NUTTX)
+    add_subdirectory(examples/slave)
```

Comment:
can we add another subfolder called nuttx (`examples/slave/nuttx`) because firstly we can possibly have slaves that do not run on nuttx. Secondly the master stack should be able to work in nuttx (unless i am wrong?). Therefore, it is slightly weird to limit ourselves to having master running only on a Linux. Techincally nuttx is almost UNIX (or at least it incorporates standard apis from unix).

Reply:
At the moment the master's mains (examples, simulation, tools) don't compile on Nuttx for various reasons.
I don't plan to make them compatible in this PR.

Reply:
I added the intermediate folder: ` examples/slave/nuttx/lan9252` as you suggested